packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -523,6 +530,43 @@ export abstract class FluidDataStoreContext
 		return factory;
 	}
 
+	createChildDataStoreSync<T extends IFluidDataStoreFactory>(
```

Comment:
What's the reasoning for:
1. Putting this on FluidDataStoreContext instead of ContainerRuntime, given that ContainerRuntime currently has the async `createDataStore`?
2. Requiring it to be a child?

Reply:
Only the context has access to a factory synchronously, as its factory must be loaded before it is realized. This is basically the same pattern we use for dds.